File: discord-bot/commands/equip.js
const { SlashCommandBuilder } = require('discord.js');
const db = require('../util/database');
const { simple } = require('../src/utils/embedBuilder');

module.exports = {
  data: new SlashCommandBuilder()
    .setName('equip')
    .setDescription('Equip an owned item')
    .addStringOption(opt =>
      opt.setName('type')
        .setDescription('Item type')
        .setRequired(true)
        .addChoices(
          { name: 'Weapon', value: 'weapon' },
          { name: 'Armor', value: 'armor' },
          { name: 'Ability', value: 'ability' }
        )
    )
    .addIntegerOption(opt =>
      opt.setName('id')
        .setDescription('Item ID')
        .setRequired(true)
    ),
  async execute(interaction) {
    const type = interaction.options.getString('type');
    const itemId = interaction.options.getInteger('id');
    const discordId = interaction.user.id;

    const { rows: players } = await db.query(
      'SELECT id FROM players WHERE discord_id = ?',
      [discordId]
    );
    if (players.length === 0) {
      const embed = simple('Character not found.');
      await interaction.reply({ embeds: [embed], ephemeral: true });
      return;
    }
    const playerId = players[0].id;

    let table;
    let column;
    if (type === 'weapon') {
      table = 'user_weapons';
      column = 'equipped_weapon_id';
    } else if (type === 'armor') {
      table = 'user_armors';
      column = 'equipped_armor_id';
    } else {
      table = 'user_ability_cards';
      column = 'equipped_ability_id';
    }

    const { rows: items } = await db.query(
      `SELECT id FROM ${table} WHERE id = ? AND player_id = ?`,
      [itemId, playerId]
    );
    if (items.length === 0) {
      const embed = simple('You do not own that item.');
      await interaction.reply({ embeds: [embed], ephemeral: true });
      return;
    }

    await db.query(
      `UPDATE players SET ${column} = ? WHERE id = ?`,
      [itemId, playerId]
    );

    const embed = simple('Item equipped!');
    await interaction.reply({ embeds: [embed], ephemeral: true });
  }
};

File: ironaccord_bot/cogs/equip.py
import discord
from discord import app_commands
from discord.ext import commands

from ..models import database as db
from ..utils.embed import simple

class EquipCog(commands.Cog):
    def __init__(self, bot: commands.Bot):
        self.bot = bot

    @app_commands.command(name="equip", description="Equip an item by ID")
    @app_commands.describe(item_id="ID of the item", type="Item type")
    @app_commands.choices(type=[
        app_commands.Choice(name="Weapon", value="weapon"),
        app_commands.Choice(name="Armor", value="armor"),
        app_commands.Choice(name="Ability", value="ability")
    ])
    async def equip(self, interaction: discord.Interaction, type: app_commands.Choice[str], item_id: int):
        player_rows = await db.query('SELECT id FROM players WHERE discord_id = %s', [str(interaction.user.id)])
        if not player_rows['rows']:
            await interaction.response.send_message(embed=simple('You have no character.'), ephemeral=True)
            return
        player_id = player_rows['rows'][0]['id']
        table = 'user_weapons' if type.value == 'weapon' else 'user_armors' if type.value == 'armor' else 'user_ability_cards'
        column = 'equipped_weapon_id' if type.value == 'weapon' else 'equipped_armor_id' if type.value == 'armor' else 'equipped_ability_id'
        owned = await db.query(f'SELECT name FROM {table} WHERE id = %s AND player_id = %s', [item_id, player_id])
        if not owned['rows']:
            await interaction.response.send_message(embed=simple('You do not own that item.'), ephemeral=True)
            return
        await db.query(f'UPDATE players SET {column} = %s WHERE id = %s', [item_id, player_id])
        await interaction.response.send_message(embed=simple(f"Equipped {owned['rows'][0]['name']}!"), ephemeral=True)

async def setup(bot: commands.Bot):
    await bot.add_cog(EquipCog(bot))

